torch/fx/experimental/rewriter.py

```diff
@@ -34,7 +34,7 @@ def rewrite(self, fn: FunctionType):
 
         # Pull out the compiled fucntion from the newly-created Module
         code = compile(dest_ast, "", "exec")
-        globals_dict = copy.copy(fn.__globals__)
+        globals_dict = fn.__globals__
```

Comment:
So instead of changing this line you want to put `fn_compiled.__globals__` down below line 42. Copying the globals here is fine and we just do it so that we can extract out the `forward`. What we want to do is to make it so that the function we ultimately return has the same globals as the input function.

Reply:
Thanks James. I have tried:
```
        globals_dict = copy.copy(fn.__globals__)
        ...
        fn_compiled.__globals__ = fn.__globals__
```

but I get a `readonly attribute error`:
```
  File "/home/melhoushi/local/miniconda3/envs/pytorch/lib/python3.9/site-packages/torch/fx/experimental/rewriter.py", line 45, in rewrite
    fn_compiled.__globals__ = fn.__globals__
AttributeError: readonly attribute
```.

Also, my original method is not working well either:
```
        globals_dict = fn.__globals__ 
        ....
```
results in the `assert len(new_keys)==1` failing for many test cases in `test_fx.py`.

Reply:
I tried to create a new class that inherits from `FunctionType` and add a `set_globals(...)` function but it seems that `FunctionType` class is set by Python to be "final" and can't be overwritten. 

Will continue to investigate other methods to solve that.

Reply:
Thanks @jamesr66a , the first method you pointed out to worked. I implemented it in this commit: e2468c7